process/smartContract/process.go

```diff
@@ -819,6 +836,40 @@ func (sc *scProcessor) processVMOutput(
 	return scrTxs, consumedFee, nil
 }
 
+func (sc *scProcessor) isTooMuchGasPutInsideTx(
+	gasRemaining uint64,
+	tx data.TransactionHandler,
+	acntSnd state.UserAccountHandler,
+) bool {
+	gasPut := big.NewInt(0)
+	gasPut.Mul(big.NewInt(0).SetUint64(tx.GetGasLimit()), big.NewInt(0).SetUint64(tx.GetGasPrice()))
+
+	if check.IfNil(acntSnd) {
+		// cross shard move balance fee was already consumed at sender shard
+		moveBalanceCost := sc.economicsFee.ComputeFee(tx)
+		gasPut.Sub(gasPut, moveBalanceCost)
+	}
+
+	gasRefunded := big.NewInt(0)
+	gasRefunded.Mul(big.NewInt(0).SetUint64(gasRemaining), big.NewInt(0).SetUint64(tx.GetGasPrice()))
```

Comment:
no need to multiplicate both provided gas and remaining gas with gasPrice. just make the difference between those 2. Plus instead of using tx as an input here, take vmInput - as it ca corrected GAS inside

Reply:
Done